packages/react/src/components/ThemeProvider/index.tsx

```diff
@@ -1,10 +1,18 @@
 import * as React from 'react';
-import { DirectionProvider } from '@radix-ui/react-direction';
+import * as RadixDirection from '@radix-ui/react-direction';
 
 import { createTheme, Theme, WebTheme } from '@aws-amplify/ui';
 
 import { AmplifyContext } from './AmplifyContext';
 
+// Radix packages don't support ESM in Node, in some scenarios(e.g. SSR),
+// the module will be imported as CommonJS module, in which we have to reference the `default`
+let sanitizedRadixDirection = { default: undefined, ...RadixDirection };
```

Comment:
Similar to your lodash idea, maybe we could reexport the radix imports from a single file, to prevent adding this workaround to primitive/component files?

Reply:
I would extract this to a util function in the follow up PR along with lodash